cloudinit/sources/helpers/vultr.py

```diff
@@ -45,45 +45,8 @@ def get_metadata(url, timeout, retries, sec_between, agent):
 
 # Set route for metadata
 def set_route(iface):
-    # Get routes, confirm entry does not exist
-    routes = netinfo.route_info()
-
-    # If no tools exist and empty dict is returned
-    if "ipv4" not in routes:
-        return
-
-    # We only care about IPv4
-    routes = routes["ipv4"]
-
-    # Searchable list
-    dests = []
-
-    # Parse each route into a more searchable format
-    for route in routes:
-        dests.append(route["destination"])
-
-    gw_present = "100.64.0.0" in dests or "100.64.0.0/10" in dests
-    dest_present = "169.254.169.254" in dests
-
-    # If not IPv6 only (No link local)
-    # or the route is already present
-    if not gw_present or dest_present:
-        return
-
-    # Set metadata route
-    if subp.which("ip"):
-        subp.subp(
-            [
-                "ip",
-                "route",
-                "add",
-                "169.254.169.254/32",
-                "dev",
-                iface,
-            ]
-        )
-    elif subp.which("route"):
-        subp.subp(["route", "add", "-net", "169.254.169.254/32", "100.64.0.1"])
+    if subp.which("route"):
+        subp.subp(["route", "add", "169.254.169.254/32", iface])
```

Comment:
Was this removal part of the original PR? You don't need the code represented here anymore?

Is it worth checking that the route add succeeds.

Reply:
It was not, it came to light during testing this on some new hardware during this process. Turns out that gateway is not as consistent as I thought. I'm not even sure what the failure is here. There is only one OS that fails, Arch Linux, and this fix shouldn't even work (though it does). I test the DHCP against the metadata IP to avoid the potential of having a customer router on one of the other network interfaces. One would assume that the DHCP url check here would fail as that route is not added, but it doesn't. However when it proceeds to the next step to pull the metadata, it fails, as the route is not there. Hence why I'm adding this there. Its very strange behavior I can't make heads or tails of. But again, it only happens on Arch Linux.

I was originally using ip and using route as a fallback if ip wasnt available such as on BSD. Though we don't actually have Cloud-Init on BSD's yet so it's not actually been tested against that particular use case as of now.

If you have any insight on this particular issue, there is likely a better solution then what I am doing here, this feels super hacky honestly. I'm running a QA round on the new images with this version against various problematic configurations now to be sure this is going to hold. I'de be fine letting this sit until thats over. This was tested, just against the current ongoing problem servers.

Reply:
Ok, to touch base here. I have a commit stored for adding back the checks for linux and the route existing but im not convinced this needs to be solved in cloud-init for this problem as it only affects arch. The route falling off for one distro between the dhcp url check and the metadata pull feels super specific to something I'm missing in this particular distro. Im going through the cloud-init code trying to find the package im missing here. I mean this code "works" it shouldn't because you would think the subp adding an already existing route would fail with a non-zero exit code on the route command, but its working across my QA on multiple distro's/hw. This as is without the waiting commit will fail on *BSD due to the difference in the route command so it will need this waiting commit to resolve that. But again, I want to make sure that this is absolutely necessary before submitting another borked commit to you guys here. Have to apologize for that as we have a very large offering spanning a wide variety of baremetal and VM profile offerings that make this incredibly difficult to get right in a single go.

Reply:
Hmmm, sorry, no particular insight here. I agree that definitely seems strange though.

Thanks for the context. Multiple commits is no problem. Would you rather I merge this as-is, or wait until you've figured out what's happening?

Reply:
I want to take a harder look. I would really prefer to remove this then add to it. But it needs one more commit to readd the route check and linux check to prevent failures if there is no other solution.

Reply:
As I had thought, this didn't need fixed here. Turns out Arch Linux by default does not install the appropriate config file to /etc/dhclient.conf. After adding this file this issue went away. Code removed.